parallel/parallel.go

```diff
@@ -90,3 +93,24 @@ func (w *Waiter) Wait() {
 func (w *Waiter) Err() <-chan error {
 	return w.errch
 }
+
+// adjustOpenFilesLimit tries to increase the soft limit of open files.
+func adjustOpenFilesLimit() {
+	var rLimit syscall.Rlimit
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
+	if err != nil {
+		return
+	}
+
+	if rLimit.Cur >= minOpenFilesLimit {
+		return
+	}
+
+	if rLimit.Max < minOpenFilesLimit {
```

Comment:
do we need this check?

Reply:
If the hard limit is lower then 1000, it returns `invalid argument` on Setrlimit() call.

Reply:
thanks